src/components/search/Filter/Filter.tsx

```diff
@@ -181,8 +248,23 @@ function Filter({
           variant="primary"
           data-testid="filter-modal-button-apply"
           onClick={() => {
-            toggleFacets(selectedFilters)
-            onDismissTransition?.()
+            // Remove facets that user unchecked
+            facetsToRemove.length > 0 && toggleFacets(facetsToRemove)
+
+            // Only toggle new facets and keep the current ones applied
+            const facetsToToggle = selectedFacets
+              .map(
+                (filter) =>
+                  !searchState.selectedFacets.includes(filter) && filter
+              )
+              .filter((m) => typeof m !== 'boolean') as IStoreSelectedFacet[]
+
+            toggleFacets(facetsToToggle)
```

Comment:
What do you think about some renaming to improve readability/semantics? Here's a suggestion:
```suggestion
            const facetsToAdd = selectedFacets
              .map(
                (facet) =>
                  !searchState.selectedFacets.includes(facet) && facet
              )
              .filter((m) => typeof m !== 'boolean') as IStoreSelectedFacet[]

            toggleFacets(facetsToAdd)
```